casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -85,33 +88,30 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
   private def computeEffect(runtime: Runtime[F], reducer: ChargingReducer[F])(
       deploy: DeployData
   ): F[EvaluateResult] =
-    runtime.deployParametersRef.set(ProtoUtil.getRholangDeployParams(deploy)) >>
-      doInj(deploy, reducer, runtime.errorLog)(runtime.cost)
-
-  /**
-    * @note `replayEval` does not need to reset the evaluation store,
-    *       merely the replay store. Hence, `replayComputeState` uses
-    *       `withRuntime` rather than `withResetRuntime`.
-    */
-  def replayComputeState(hash: StateHash)(
+    for {
+      _      <- runtime.deployParametersRef.set(ProtoUtil.getRholangDeployParams(deploy))
+      result <- doInj(deploy, reducer, runtime.errorLog)(runtime.cost)
+    } yield result
+
+  def replayComputeState(startHash: StateHash)(
       terms: Seq[InternalProcessedDeploy],
       blockTime: Long
   ): F[Either[ReplayFailure, StateHash]] =
-    withRuntime { runtime =>
+    withRuntimeLock { runtime =>
       for {
         _      <- setBlockTime(blockTime, runtime)
-        result <- replayDeploys(hash, terms, replayDeploy(runtime))
+        result <- replayDeploys(startHash, terms, replayDeploy(runtime))
       } yield result
     }
 
-  def computeState(hash: StateHash)(
+  def computeState(startHash: StateHash)(
       terms: Seq[DeployData],
       blockTime: Long
   ): F[(StateHash, Seq[InternalProcessedDeploy])] =
-    withResetRuntime(hash) { runtime =>
+    withRuntimeLock { runtime =>
```

Comment:
Can you say a few words about this change? Why is the "reset" part not necessary anymore?

Reply:
It's in the commit message :)

Reply:
processDeploys does `reset` as its first thing